### PR TITLE
Add handling for TEMP and GSR channel types

### LINKED ISSUE #1046: BIDS channel type issue
bids_path = BIDSPath(root='../ds003061')

Channels 65 to 79 are not EEG (and marked as such in the BIDS repo). Yet MNE thinks 2 of these channels are EEG.

https://openneuro.org/datasets/ds003061

### PR DESCRIPTION
**Pull request by hoechenberger**
Fixes #1046

**Please somebody feel free to take over and finalize this. It still needs a test.**

Merge checklist
---------------

Maintainer, please confirm the following before merging.
If applicable:

- [ ] All comments are resolved
- [ ] This is not your own PR
- [ ] All CIs are happy
- [ ] PR title starts with [MRG]
- [ ] [whats_new.rst](https://github.com/mne-tools/mne-bids/blob/main/doc/whats_new.rst) is updated
- [ ] New contributors have been added to [CITATION.cff](https://github.com/mne-tools/mne-bids/blob/main/CITATION.cff)
- [ ] PR description includes phrase "closes <#issue-number>"

### CODE REVIEW

File: doc/whats_new.rst [hunk 1]
```diff
@@ -92,6 +92,8 @@ Detailed list of changes
 
 - Instead of deleting files and raising cryptic errors, an intentional error message is now sent when calling :func:`~mne_bids.write_raw_bids` with the source file identical to the destination file, unless ``format`` is specified, by `Adam Li`_ and `Stefan Appelhoff`_ (:gh:`889`)
 
+- Correctly handle "TEMP" and "GSR" channel types in :func:`~mne_bids.write_raw_bids`, by `Richard Höchenberger`_ (:gh:`xxx`)
```
Comment (hoechenberger) [inline doc/whats_new.rst:95]:
```suggestion
- Correctly handle "TEMP" and "GSR" channel types in :func:`~mne_bids.read_raw_bids`, by `Richard Höchenberger`_ (:gh:`xxx`)
```

File: mne_bids/utils.py [hunk 2]
```diff
@@ -76,11 +76,14 @@ def _get_ch_type_mapping(fro='mne', to='bids'):
...
     elif fro == 'bids' and to == 'mne':
         mapping = dict(EEG='eeg', MISC='misc', TRIG='stim', EMG='emg',
-                       ECOG='ecog', SEEG='seeg', EOG='eog', ECG='ecg',
+                       ECOG='ecog', SEEG='seeg', ECG='ecg',
                        RESP='resp', NIRS='fnirs_cw_amplitude',
+                       DBS='dbs',
                        # No MEG channels for now
-                       # Many to one mapping
-                       VEOG='eog', HEOG='eog', DBS='dbs')
+                       # Many to one mappings
```
Comment (hoechenberger) [inline mne_bids/utils.py:83]:
```suggestion
                       # Many-to-one mappings
```

File: mne_bids/utils.py [hunk 3]
```diff
@@ -76,11 +76,13 @@ def _get_ch_type_mapping(fro='mne', to='bids'):
...
-                       ECOG='ecog', SEEG='seeg', EOG='eog', ECG='ecg',
+                       ECOG='ecog', SEEG='seeg', ECG='ecg',
                        RESP='resp', NIRS='fnirs_cw_amplitude',
+                       DBS='dbs',
                        # No MEG channels for now
-                       # Many to one mapping
-                       VEOG='eog', HEOG='eog', DBS='dbs')
+                       # Many-to-one mappings
+                       EOG='eog', VEOG='eog', HEOG='eog',
+                       TEMP='misc', GSR='misc',)
```
Comment (agramfort) [inline mne_bids/utils.py:85]: to me TEMP and GSR are not part of the standard

this data contains channels with type "temp" and "GSR"

Cf https://bids-specification.readthedocs.io/en/stable/04-modality-specific-files/03-electroencephalography.html#channels-description-_channelstsv

Type of channel; MUST use the channel types listed below. Note that the type MUST be in upper-case.  

Must be one of: "MEGMAG", "MEGGRADAXIAL", "MEGGRADPLANAR", "MEGREFMAG", "MEGREFGRADAXIAL", "MEGREFGRADPLANAR", "MEGOTHER", "EEG", "ECOG", "SEEG", "DBS", "VEOG", "HEOG", "EOG", "ECG", "EMG", "TRIG", "AUDIO", "PD", "EYEGAZE", "PUPIL", "MISC", "SYSCLOCK", "ADC", "DAC", "HLU", "FITERR", "OTHER".

Comment (hoechenberger) [reply mne_bids/utils.py:85]: The list is outdated. Scroll down a little more to see a table with all supported types. The list should be removed and instead a reference to the table should be added.

**Comment by agramfort** [comment]
arfff

I guess we should add these channel types in MNE then...

**Comment by agramfort** [comment]
I would say yes...

**Comment by hoechenberger** [comment]
Superseded by #1052

**Pull request closed without merging**